ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsNoJobTask.java

```diff
@@ -90,12 +91,27 @@ public BasicStatsNoJobTask(HiveConf conf, BasicStatsNoJobWork work) {
     console = new LogHelper(LOG);
   }
 
-  public static boolean canUseFooterScan(
+  public static boolean canUseStats(
       Table table, Class<? extends InputFormat> inputFormat) {
+      return (OrcInputFormat.class.isAssignableFrom(inputFormat) && !AcidUtils.isFullAcidTable(table))
+          || MapredParquetInputFormat.class.isAssignableFrom(inputFormat)
+          || useBasicStatsFromStorageHandler(table);
+  }
+
+  public static boolean canUseColumnStats(Table table, Class<? extends InputFormat> inputFormat) {
     return (OrcInputFormat.class.isAssignableFrom(inputFormat) && !AcidUtils.isFullAcidTable(table))
         || MapredParquetInputFormat.class.isAssignableFrom(inputFormat);
   }
 
+  private static boolean useBasicStatsFromStorageHandler(Table table) {
+    if (table.isNonNative()) {
+      TableDesc tableDesc = Utilities.getTableDesc(table);
+      return table.getStorageHandler().getBasicStatistics(tableDesc) != null;
```

Comment:
Aren't we calculating all the stats here by calling `getBasicStatistics`, just to then discard the results?

Reply:
Introduced a new method on the interface to check whether the storage handler can provide stats.